linker-interface/shared/src/main/scala/org/scalajs/linker/interface/Fingerprint.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+package org.scalajs.linker.interface
+
+import scala.collection.mutable
+
+private[interface] trait Fingerprint[T] {
+  def fingerprint(obj: T): String
+}
+
+private[interface] object Fingerprint {
+  def fingerprint[T: Fingerprint](obj: T): String =
+    implicitly[Fingerprint[T]].fingerprint(obj)
+
+  class FingerprintBuilder(name: String) {
+    private val fields = new mutable.ListBuffer[String]
+
+    def addField[T: Fingerprint](name: String, value: T): FingerprintBuilder = {
+      fields.append(s"$name=${fingerprint(value)}")
+      this
+    }
+
+    def build(): String = fields.mkString(s"$name(", ",", ")")
+  }
+
+  implicit object BooleanFingerprint extends Fingerprint[Boolean] {
+    override def fingerprint(bool: Boolean): String = bool.toString
+  }
+
+  implicit object IntFingerprint extends Fingerprint[Int] {
+    override def fingerprint(int: Int): String = int.toString
+  }
+
+  implicit object StringFingerprint extends Fingerprint[String] {
+    override def fingerprint(str: String): String = str
+  }
+
+  implicit def optionFingerprint[T](implicit itemFingerprint: Fingerprint[T]): Fingerprint[Option[T]] =
+    new Fingerprint[Option[T]] {
+      override def fingerprint(opt: Option[T]): String =
+        opt.map(itemFingerprint.fingerprint).map(f => s"Some($f)").getOrElse("None")
```

Comment:
Use `fold` instead of `map` followed by `getOrElse`.

Reply:
Fixed in 5628647d30f13ee91950c380e1bbf5ea597acd01